src/pymor/models/transfer_function.py

```diff
@@ -120,62 +121,103 @@ def eval_dtf(self, s, mu=None):
         else:
             return self.dtf(s, mu=mu)
 
-    def freq_resp(self, w, mu=None):
+    @cached
+    def freq_resp(self, w=None, mu=None, adaptive_type='bode', adaptive_opts=None):
         """Evaluate the transfer function on the imaginary axis.
 
         Parameters
         ----------
         w
-            A sequence of angular frequencies at which to compute the transfer function.
+            If `len(w) == 2`, the left and right limits used for the adaptive sampling.
+            Otherwise, a sequence of angular frequencies at which to compute the transfer function.
         mu
             |Parameter values| for which to evaluate the transfer function.
+        adaptive_type
+            The plot type that adaptive sampling should be used for
+            (`'bode'`, `'mag'`).
+            Ignored if `len(w) != 2`.
```

Comment:
My reason for this was to make it easier to extend to new types in the future, e.g., for the sigma or Nyquist plots. Also, I would like to make the `adaptive` function more or less independent of transfer functions (I imagine that it could be used for other things).

I realize now that actually `np.abs` should be replaced with a `norm`...